subsys/bluetooth/host/scan.h

```diff
@@ -5,12 +5,123 @@
 
 int bt_le_scan_set_enable(uint8_t enable);
 
 struct bt_le_per_adv_sync *bt_hci_get_per_adv_sync(uint16_t handle);
 
 void bt_periodic_sync_disable(void);
+
+/**
+ * Start / update the scanner.
+ *
+ * This API updates the users of the scanner.
+ * Multiple users can be enabled at the same time.
+ * Depending on all the users, scan parameters are selected
+ * and the scanner is started or updated, if needed.
+ * This API may update the scan parameters, for example if the scanner is already running
+ * when another user that demands higher duty-cycle is being added.
+ * It is not allowed to add a user that was already added.
+ *
+ * Every SW module that informs the scanner that it should run, needs to eventually remove
+ * the flag again using @ref bt_le_scan_user_remove once it does not require
+ * the scanner to run, anymore.
+ *
+ * If flag is set to @ref BT_LE_SCAN_USER_NONE, no user is being added. Instead, the
+ * existing users are checked and the scanner is started, stopped or updated.
+ * For platforms where scanning and initiating at the same time is not supported,
+ * this allows the background scanner to be started or stopped once the device starts to
+ * initiate a connection.
+ *
+ * @param flag user requesting the scanner
+ *
+ * @retval 0 in case of success
+ * @retval -EALREADY if the role is already enabled
+ * @retval -EPERM    if the explicit scanner is being enabled while the initiator is running
+ *                    and the device does not support this configuration.
+ * @retval -ENOBUFS  if no hci command buffer could be allocated
+ * @retval -EBUSY    if the scanner is updated in a different thread. The user was added but
+ *                   the scanner was not started/stopped/updated.
+ * @returns negative error codes for errors in @ref bt_hci_cmd_send_sync
+ */
+int bt_le_scan_user_add(enum bt_le_scan_enabled_flag flag);
+
+/**
+ * Stop / update the scanner.
+ *
+ * This API updates the users of the scanner.
+ * Depending on all enabled users, scan parameters are selected
+ * and the scanner is stopped or updated, if needed.
+ * This API may update the scan parameters, for example if the scanner is already running
+ * when a user that demands higher duty-cycle is being removed.
+ * Removing a role that was not added does not result in an error.
+ *
+ * This API allows removing the user why the scanner is running.
+ * If all users for the scanner to run are removed, this API will stop the scanner.
+ *
+ * If flag is set to @ref BT_LE_SCAN_USER_NONE, no user is being added. Instead, the
+ * existing users are checked and the scanner is started, stopped or updated.
+ * For platforms where scanning and initiating at the same time is not supported,
+ * this allows the background scanner to be started or stopped once the device starts to
+ * initiate a connection.
+ *
+ * @param flag user releasing the scanner
+ *
+ * @retval 0 in case of success
+ * @retval -EALREADY if the role is already enabled
```

Comment:
```suggestion
 * @retval -EALREADY if the user is not added
```
or
```suggestion
 * @retval -EALREADY if the user is not enabled
```

Reply:
ah actually that error code should not be there